php-frontend/src/main/java/org/sonar/plugins/php/api/tests/PHPCheckVerifier.java

```diff
@@ -0,0 +1,126 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php.api.tests;
+
+import com.google.common.base.Preconditions;
+import com.sonar.sslr.api.typed.ActionParser;
+import com.sonarsource.checks.verifier.SingleFileVerifier;
+import java.io.File;
+import org.sonar.php.parser.PHPParserBuilder;
+import org.sonar.php.tree.visitors.LegacyIssue;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxTrivia;
+import org.sonar.plugins.php.api.visitors.FileIssue;
+import org.sonar.plugins.php.api.visitors.IssueLocation;
+import org.sonar.plugins.php.api.visitors.LineIssue;
+import org.sonar.plugins.php.api.visitors.PHPCheck;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+import org.sonar.plugins.php.api.visitors.PhpFile;
+import org.sonar.plugins.php.api.visitors.PhpIssue;
+import org.sonar.plugins.php.api.visitors.PreciseIssue;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class PHPCheckVerifier {
+
+  private static final ActionParser<Tree> parser = PHPParserBuilder.createParser();
+
+  public static void verify(File sourceFile, PHPCheck check) {
+    verify(new PhpTestFile(sourceFile), check);
+  }
+
+  public static void verifyNoIssue(File sourceFile, PHPCheck check) {
+    verifyNoIssue(new PhpTestFile(sourceFile), check);
+  }
+
+  public static void verify(PhpFile sourceFile, PHPCheck check) {
+    createVerifier(sourceFile, check).assertOneOrMoreIssues();
+  }
+
+  public static void verifyNoIssue(PhpFile sourceFile, PHPCheck check) {
+    createVerifier(sourceFile, check).assertNoIssues();
+  }
+
+  private static SingleFileVerifier createVerifier(PhpFile file, PHPCheck check) {
+    SingleFileVerifier verifier = SingleFileVerifier.create(file.relativePath(), UTF_8);
+    CompilationUnitTree tree = (CompilationUnitTree) parser.parse(file.contents());
+    check.init();
+    for (PhpIssue issue : check.analyze(file, tree)) {
+      Preconditions.checkState(issue.check().equals(check), "Verifier support only one kind of issue " + issue.check() + " != " + check);
+      addIssue(verifier, check, issue).withGap(issue.cost());
+    }
+    PHPVisitorCheck commentVisitor = new PHPVisitorCheck() {
+      @Override
+      public void visitTrivia(SyntaxTrivia trivia) {
+        super.visitTrivia(trivia);
+        int suffixLength = trivia.text().startsWith("//") ? 0 : 2;
+        verifier.addComment(trivia.line(), trivia.column() + 1, trivia.text(), 2, suffixLength);
+      }
+    };
+    commentVisitor.analyze(file, tree);
+    return verifier;
+  }
+
+  private static SingleFileVerifier.Issue addIssue(SingleFileVerifier verifier, PHPCheck check, PhpIssue issue) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "check". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)